src/sentry/eventstore/utils.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import
+
+import logging
+
+methods_to_test = [
+    "get_next_event_id",
+    "get_prev_event_id",
+    "get_earliest_event_id",
+    "get_latest_event_id",
+]
+
+logger = logging.getLogger("sentry.eventstore")
+
+
+def selector_func(context, method, callargs):
+    if method in methods_to_test:
+        return ["snuba", "snuba_discover"]
+
+    return ["snuba"]
+
+
+def callback_func(context, method, callargs, backends, results):
+    """
+    Log if results are different
+    """
+    if backends == ["snuba", "snuba_discover"] and method in methods_to_test:
+        if results[0].result() != results[1].result():
+            logger.info(
+                "discover.result-mismatch",
+                extra={
+                    "method": method,
+                    "callargs": callargs,
+                    "snuba": results[0].result(),
+                    "snuba_discover": results[1].result(),
```

Comment:
please ensure you do not flood the logs in case result is big.
Either show only the differences or put a limit on the size of what you are showing maybe by only showing event ids.

Reply:
We'll need to change this for the `get_events` and `get_event_by_id` methods since they return actual event objects. For the methods under test right now it's only the event_id and project_id that's returned.

Reply:
Still the result could be quite big, could you please either log the differences or hard limit the result size? (the first is better)

Reply:
Ok. I'll also update the callargs since an Event might be largs

Reply:
Updated the callargs to be smaller. The result cannot be big. It's only a tuple of `(project_id, event_id)` or `None`